<launch>
	<arg name="p_num" default="4"/>
	<arg name="veh" default="master"/>
	<arg name="local" default="true"/>
    <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />
    <arg name="joy_mapper_param_file_name" default="$(arg param_file_name)" doc="Specify a joy_mapper param file. ex:high_speed" />

	<!-- joy -->
    <node ns="$(arg veh)" if="$(arg local)" pkg="joy" type="joy_node" name="joy" output="screen">
        <rosparam command="load" file="$(find duckietown)/config/$(arg config)/joy/joy_node/$(arg param_file_name).yaml"/>
    </node>

	<!-- start node node -->
	<include file="$(find master)/launch/master_name_node.launch">
		<arg name="veh" value="$(arg veh)"/>
		<arg name="param_file_name" default="default" />
	</include>

	<!-- start basic args -->
	<remap from="patrolling_node/rm_robot" to="master_name_node/rm_robot"/>
	<remap from="patrolling_node/print_cost" to="master_name_node/print_cost"/>
	<remap from="patrolling_node/print_state" to="master_name_node/print_state"/>
	<remap from="patrolling_node/setpub" to="master_name_node/setpub"/>
	<!-- remap from="patrolling_node/reset" to="master_name_node/reset"/-->
	<include file="$(find patrolling_node)/launch/patrolling_node.launch">
		<arg name="p_num" value="$(arg p_num)"/>
		<arg name="veh" value="$(arg veh)"/>
		<arg name="local" value="$(arg local)"/>
	</include>

</launch>
